### PR TITLE
feat: add encrypt user mail to userdata Datalayer

### PR DESCRIPTION
**Pull request by guilhermesennes**
#### What is the purpose of this pull request?

Add the user's encrypted email to the 'userData' data layer.

#### What problem is this solving?

We need to capture data for Google to increase conversions in Google campaigns. It's about optimized conversions. It is necessary to start receiving the email of the logged in user encrypted in SHA256, that is, an anonymized source data, with this, we will not have problems in terms of LGPD

#### How should this be manually tested?

1. Access Workspace;
2. Log in;
3. In the console, verify that when typing dataLayer, in the 'userData' layer, the 'cryptId' field is displayed, which refers to the user's encrypted email.

#### Screenshots or example usage

image

#### Types of changes

* [ ] Bug fix (a non-breaking change which fixes an issue)
* [X] New feature (a non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/modules/extraEvents.ts [hunk 1]
```diff
@@ -20,13 +29,16 @@ export async function sendExtraEvents(e: PixelMessage) {
     case 'vtex:userData': {
       const { data } = e
 
+      const criptoID = await digestMessage(data.email)
+
       if (!data.isAuthenticated) {
         return
       }
```
Comment (icazevedo) [inline react/modules/extraEvents.ts:36]: Shouldn't we move it past the `isAuthenticated` check? Is there any reason you put it before?
I also think we should be aware that email can be `null`/`undefined` for a number of reasons, and that would make a hash to be generated when it shouldn't. We should prepare for that.

```suggestion
      if (!data.isAuthenticated) {
        return
      }
      
      const criptoID = data.email ? await digestMessage(data.email) : undefined
```

**Review by icazevedo — CHANGES_REQUESTED** [review]
I left a change in the comments. Also, include a changelog entry describing the changes you've made and why.

File: react/modules/extraEvents.ts (hunk 1 above)
Comment (guilhermesennes) [reply react/modules/extraEvents.ts]: @icazevedo Good afternoon.
I made the necessary changes. Excellent observation, thank you.

File: react/modules/extraEvents.ts [hunk 2]
```diff
@@ -2,6 +2,15 @@ import push from './push'
 import { PixelMessage } from '../typings/events'
 
 export async function sendExtraEvents(e: PixelMessage) {
+
+  async function digestMessage(message:string) {
```
Comment (icazevedo) [inline react/modules/extraEvents.ts:6]:
```suggestion
  async function emailToHash(email:string) {
```

**Review by icazevedo — CHANGES_REQUESTED** [review]
Left some minor changes. I also think we can improve the naming of the variables and functions to easily identify what they are and what they do.

File: react/modules/extraEvents.ts [hunk 3]
```diff
@@ -2,6 +2,15 @@ import push from './push'
...
+    const msgUint8 = new TextEncoder().encode(message);                           
+    const hashBuffer = await crypto.subtle.digest('SHA-256', msgUint8);
+    const hashArray = Array.from(new Uint8Array(hashBuffer));
+    const hashHex = hashArray.map(b => b.toString(16).padStart(2, '0')).join('');
+    return hashHex;
+  }
```
[the remark this answers is not in the document]
Comment (guilhermesennes) [reply react/modules/extraEvents.ts]: > Left some minor changes. I also think we can improve the naming of the variables and functions to easily identify what they are and what they do.

Good afternoon, @icazevedo!

I made the changes you've said. Can you validate please?

**Comment by perillosantana** [comment]
@icazevedo

**Comment by guilhermesennes** [comment]
@icazevedo

**Review by icazevedo — APPROVED** [review]

**Comment by icazevedo** [comment]
@guilhermesennes please rebase your branch so we can merge it

**Pull request merged**